### PR TITLE
Wasm executor support

### PR DESCRIPTION
**Pull request by lulf**

### CODE REVIEW

File: embassy/src/executor/arch/wasm.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /// for example by passing it as an argument to the initial tasks.
+    ///
+    /// This function requires `&'static mut self`. This means you have to store the
+    /// Executor instance in a place where it'll live forever and grants you mutable
+    /// access. There's a few ways to do this:
+    ///
+    /// - a [Forever](crate::util::Forever) (safe)
+    /// - a `static mut` (unsafe)
+    /// - a local variable in a function you know never returns (like `fn main() -> !`), upgrading its lifetime with `transmute`. (unsafe)
+    pub fn start(&'static mut self, init: impl FnOnce(Spawner) + Send) {
```
Comment (Dirbaio) [inline embassy/src/executor/arch/wasm.rs:65]: Why is `Send` required on `init`?

File: embassy/src/executor/raw/mod.rs [hunk 2]
```diff
@@ -10,7 +10,7 @@
 mod run_queue;
 #[cfg(feature = "time")]
 mod timer_queue;
-mod util;
+pub(crate) mod util;
```
Comment (Dirbaio) [inline embassy/src/executor/raw/mod.rs:13]: I think this pub is not used?

**Comment by Dirbaio** [comment]
This is awesome! :D 

Could you add building the WASM example to CI, so that it doesn't bitrot?

File: embassy/src/time/mod.rs [hunk 3]
```diff
@@ -51,6 +51,9 @@ mod timer;
 #[cfg(feature = "std")]
 mod driver_std;
 
+#[cfg(feature = "wasm")]
+mod driver_wasm;
```
Comment (Dirbaio) [inline embassy/src/time/mod.rs:55]: I think this file is missing

File: embassy/src/executor/arch/wasm.rs (hunk 1 above)
Comment (lulf) [reply embassy/src/executor/arch/wasm.rs:65]: Copy paste from some other executor :)

File: embassy/src/executor/raw/mod.rs (hunk 2 above)
Comment (lulf) [reply embassy/src/executor/raw/mod.rs:13]: Ah, yes. I tried to use the UninitCell, but discovered it did not implement Sync + Send.

**Comment by lulf** [comment]
@Dirbaio Also needs https://github.com/embassy-rs/critical-section/pull/2 right now, not sure if there is another workaround you can think of?

**Review by Dirbaio — APPROVED** [review]
Thank you! :)

**Pull request merged**